src/util/ActionProgramUtils.ts

```diff
@@ -110,42 +110,60 @@ export const makeAaveDepositAction = async ({
     walletId: borrowEngineWallet.id
   })
 
-  // @ts-expect-error
   return out
 }
 
-export const makeAaveRepayAction = async (borrowPluginId: string, nativeAmount: string, tokenId: string, wallet: EdgeCurrencyWallet) => {
-  // TODO: Spec out behavior for potentially repaying with a different asset.
-  const repayToken = getToken(wallet, tokenId)
-  if (repayToken == null) throw new Error(`Could not find repayment token ${tokenId} on ${wallet.currencyInfo.currencyCode} wallet`)
-  await enableToken(repayToken.currencyCode, wallet)
-  return {
-    type: 'loan-repay',
-    borrowPluginId,
-    nativeAmount,
-    tokenId,
-    walletId: wallet.id
+export const makeAaveRepayAction = async (
+  borrowPluginId: string,
+  debtTokenId: string,
+  wallet: EdgeCurrencyWallet,
+  collateralTokenId?: string,
+  nativeAmount?: string
+): Promise<ActionOp> => {
+  if (collateralTokenId != null) {
+    return {
+      type: 'loan-swap-repay',
+      borrowPluginId,
+      collateralTokenId,
+      debtTokenId,
+      walletId: wallet.id
+    }
+  } else {
+    const repayToken = getToken(wallet, debtTokenId)
+    if (repayToken == null) throw new Error(`Could not find repayment token ${debtTokenId} on ${wallet.currencyInfo.currencyCode} wallet`)
+    await enableToken(repayToken.currencyCode, wallet)
+    return {
+      type: 'loan-repay',
+      borrowPluginId,
+      nativeAmount: nativeAmount ?? ethers.constants.MaxUint256.toString(),
+      tokenId: debtTokenId,
+      walletId: wallet.id
+    }
   }
 }
 
-export const makeAaveWithdrawAction = async (borrowPluginId: string, nativeAmount: string, tokenId: string, wallet: EdgeCurrencyWallet) => {
+export const makeAaveWithdrawAction = async (borrowPluginId: string, tokenId: string, wallet: EdgeCurrencyWallet, nativeAmount?: string): Promise<ActionOp> => {
   const withdrawalToken = getToken(wallet, tokenId)
   if (withdrawalToken == null) throw new Error(`Could not find withdrawal token ${tokenId} on ${wallet.currencyInfo.currencyCode} wallet`)
   await enableToken(withdrawalToken.currencyCode, wallet)
   return {
     type: 'loan-withdraw',
     borrowPluginId,
-    nativeAmount,
+    nativeAmount: nativeAmount ?? ethers.constants.MaxUint256.toString(),
     tokenId,
     walletId: wallet.id
   }
 }
 
-export async function makeActionProgram(actionOp: ActionOp, id?: string): Promise<ActionProgram> {
-  const programId = Date.now().toString() + (id ?? '')
-
-  return {
-    programId,
-    actionOp
-  }
+export const makeAaveCloseAction = async (
+  borrowPluginId: string,
+  collateralTokenId: string,
+  debtTokenId: string,
+  wallet: EdgeCurrencyWallet
+): Promise<ActionOp[]> => {
+  const out = await Promise.all([
+    makeAaveRepayAction(borrowPluginId, debtTokenId, wallet, collateralTokenId),
+    makeAaveWithdrawAction(borrowPluginId, collateralTokenId, wallet)
+  ])
```

Comment:
The amount needs to be specified here because the withdraw Action display shows the amount